### PR TITLE
Add urllib3.response stubs

### PR DESCRIPTION
**Pull request by jpy-git**
Updates the `urllib3.response` stubs inline with the latest release:
https://github.com/urllib3/urllib3/blob/b1f60e44d43b13e5272d5b6003f125af9c25c8ad/src/urllib3/response.py

### CODE REVIEW

File: stubs/urllib3/urllib3/response.pyi [hunk 1]
```diff
@@ -1,66 +1,96 @@
...
 class GzipDecoder:
     def __init__(self) -> None: ...
-    def __getattr__(self, name): ...
-    def decompress(self, data): ...
+    def __getattr__(self, name: str) -> Any: ...
+    def decompress(self, data: bytes) -> bytes: ...
+
+class BrotliDecoder:
+    def __init__(self) -> None: ...
+    def flush(self) -> bytes: ...
```
Comment (jpy-git) [inline stubs/urllib3/urllib3/response.pyi:28]: this class is only available if `brotli` is available for import. Wasn't sure how to indicate this in stubs so just added to the allowlist for now

File: stubs/urllib3/urllib3/response.pyi [hunk 2]
```diff
@@ -1,66 +1,96 @@
...
     def __init__(self) -> None: ...
-    def __getattr__(self, name): ...
-    def decompress(self, data): ...
+    def __getattr__(self, name: str) -> Any: ...
+    def decompress(self, data: bytes) -> bytes: ...
+
+class GzipDecoderState:
+    FIRST_MEMBER: int
+    OTHER_MEMBERS: int
+    SWALLOW_DATA: int
```
Comment (srittau) [inline stubs/urllib3/urllib3/response.pyi:19]:
```suggestion
    FIRST_MEMBER: Literal[0]
    OTHER_MEMBERS: Literal[1]
    SWALLOW_DATA: Literal[2]
```

`Literal` needs to be imported from `typing_extensions`.

File: stubs/urllib3/urllib3/response.pyi (hunk 1 above)
Comment (srittau) [reply stubs/urllib3/urllib3/response.pyi:28]: Allowlist is fine. We should also add a comment in the stub file itself.

File: stubs/urllib3/urllib3/response.pyi [hunk 3]
```diff
@@ -1,66 +1,96 @@
...
+        msg: _HttplibHTTPMessage | None = ...,
+        retries: Retry | None = ...,
+        enforce_content_length: bool = ...,
+        request_method: str | None = ...,
+        request_url: str | None = ...,
+        auto_close: bool = ...,
     ) -> None: ...
-    def get_redirect_location(self): ...
-    def release_conn(self): ...
+    def get_redirect_location(self) -> bool | str | None: ...
```
Comment (srittau) [inline stubs/urllib3/urllib3/response.pyi:71]:
```suggestion
    def get_redirect_location(self) -> Literal[False] | str | None: ...
```

File: stubs/urllib3/urllib3/response.pyi [hunk 4]
```diff
@@ -1,66 +1,96 @@
...
+    def tell(self) -> int: ...
+    def read(self, amt: int | None = ..., decode_content: bool | None = ..., cache_content: bool = ...) -> bytes: ...
+    def stream(self, amt: int | None = ..., decode_content: bool | None = ...) -> Iterator[bytes]: ...
     @classmethod
-    def from_httplib(cls, r, **response_kw): ...
-    def getheaders(self): ...
-    def getheader(self, name, default=...): ...
-    def close(self): ...
+    def from_httplib(cls, r: _HttplibHTTPResponse, **response_kw: Any) -> HTTPResponse: ...
+    def getheaders(self) -> list[tuple[str, str]]: ...
```
Comment (srittau) [inline stubs/urllib3/urllib3/response.pyi:84]:
```suggestion
    def getheaders(self) -> HTTPHeaderDict: ...
```

**Review by srittau — CHANGES_REQUESTED** [review]
Thanks, remarks below.

**Review by srittau — APPROVED** [review]

**Pull request merged**